src/api/routes/offer.js

```diff
@@ -30,12 +32,16 @@ module.exports = (app) => {
     router.post("/", authMiddleware.isGod, validators.create, async (req, res) => {
         try {
             // This is safe since the service is destructuring the passed object and the fields have been validated
-            const offer = await (new OfferService()).create(req.body);
+            if (await Offer.find().activeOffersCount(req.body).length < Offer.MAX_ACTIVE_OFFERS_ALLOWED) {
```

Comment:
This should be in the validators.create, as a custom validator, so that the error is automatically produced and standardized.

Reply:
Also, you don't want to pass all the req.body, as it has a lot of properties, use req.body.owner instead, as it represents the company

Reply:
Another thing: instead of using `.length`, use `countDocuments()`, so that the query is more efficient, since you don't need the actual offers, only how many.